### DISCUSSION TITLE
Power_Heatpump topic

### DISCUSSION THREAD

**Discussion by Axelaar** [Arnold-n/P1P2MQTT#116] [Q&A] [answered]
(I asked this question in a closed issues post, this is likely a better place to ask this question).
I updated to P1P2MQTT bridge v0.9.53 and P1P2Monitor v0.9.54 from versions 9.42. In those versions Power_Heatpump was available, as the calculated value of the heat produced. In that older version it was positive for heating, and negative for cooling.
The topic was listed under .../M/9/...  In an issue post it was mentioned that this name would be changed to Heat_Produced_Heatpump.

I am currently testing to see if in the new version the topic could be ..../C/9/Production_Heatpump, however that topic returns very large numbers that I cannot make sense of.

What is the equivalent for that topic in this new version, and the scale/unit of that number?

**Comment by Arnold-n** [comment, marked as answer]
I would have answered the closed issue too :-). The new entity is  `P1P2/P/P1P2MQTT/bridge0/C/9/Production_Heatpump` and it should report in Watt (+ for heating or - for cooling). Are you seeing anything much larger than your system's capacity?

**Reply by Axelaar** [reply]
I indeed assumed that the new entity you mentioned was the one - but it does not report numbers like before.  I feed the number into my visualization system where it is handled as a 16-bit EIS5 number.  That number maxes out at 671088, it appears to be at least 1000 times the expected number.

**Reply by Arnold-n** [reply]
Strange, it should report ROUND(69.7 * flow * dT) as a 16-bit signed integer. What does it report and what do the following topics report on your side?
-`P1P2/P/P1P2MQTT/bridge0/M/1/Flow`
-`P1P2/P/P1P2MQTT/bridge0/Temperature_R4T_Return_Water`
-`P1P2/P/P1P2MQTT/bridge0/Temperature_R1T_HP2Gas_Water` (or alternatively `.../Temperature_R1T_Leaving_Water` if "P43: use R1T instead of R2T" would be set to 1)

**Reply by Axelaar** [reply]
Received b'25.5' from topic P1P2/P/P1P2MQTT/bridge0/M/1/Flow
Received b'17.500' from topic P1P2/P/P1P2MQTT/bridge0/T/1/Temperature_R4T_Return_Water
Received b'14.641' from topic P1P2/P/P1P2MQTT/bridge0/T/1/Temperature_R1T_HP2Gas_Water
and
Received b'-5261' from topic P1P2/P/P1P2MQTT/bridge0/C/9/Production_Heatpump
That 5261 would be the proper number.  I need to see if I have in the visualization server something that multiplied the old topic.  I did not change anything on that side.  Will look into it right away and report back.

**Reply by Axelaar** [reply]
Indeed, using the previous version P1P2 I had to multiply the number with 1000.
I will remove that multiplication.
Thanks for your feedback!

**Comment by Arnold-n** [comment]
You can delete the old unused entities with the `MQTT_Delete_All_Build` button in the `HC_Setup` device if the `HA_Setup` switch is enabled.

**Reply by Axelaar** [reply]
I am not using Home Assistant.  Indeed I still get the last seen values of the old entities when I restart my Python script.  I now filter out the old prefix which had the fourth segment of the IP address as identifier.  Would be nice to know how to delete all the old information using for instance a telnet session.

**Reply by Arnold-n** [reply]
You can delete old values with the new command `D14` which corresponds to HA button `MQTT_Delete_All_rebuild`.
